### PR TITLE
Adds responsive header

### PR DESCRIPTION
**Pull request by Moejoe90**
Large
big screen
Medium
medium
Small
mobile

### CODE REVIEW

File: app/components/Header.tsx [hunk 1]
```diff
@@ -1,26 +1,182 @@
...
+export class Header extends React.Component<IHeaderProps, IHeaderState> {
+  constructor(props: IHeaderProps) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  toggle(): any {
```
Comment (Moejoe90) [inline app/components/Header.tsx]: How would you deal with this situation what would type this?

File: app/components/Header.tsx [hunk 2]
```diff
@@ -1,26 +1,182 @@
...
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
+  }
+
+  render(): any {
```
Comment (Moejoe90) [inline app/components/Header.tsx]: same here

File: app/components/Header.tsx [hunk 3]
```diff
@@ -1,26 +1,182 @@
...
+  name: string;
+  toggle: () => {};
+  isOpen: boolean;
+}
+
+export const NeufundBrand: React.SFC<INeufundBrand> = ({ classStyle }) => {
+  return (
+    <Row>
+      <img
+        src="https://neufund.org/img/social_logo.png"
```
Comment (Moejoe90) [inline app/components/Header.tsx]: I took logo as a link for now until an assets folder is set, I know there is another branch with assets, so I didn't want to create conflict

File: app/components/Header.tsx (hunk 1 above)
Comment (krzkaczor) [reply app/components/Header.tsx]: it does not return anything so - `void`

File: app/components/Header.tsx (hunk 2 above)
Comment (krzkaczor) [reply app/components/Header.tsx]: You can read it from class that you extending - React.component (just click on it and your IDE should navigate) `React.ReactNode` is correct type

File: app/components/Header.tsx [hunk 4]
```diff
@@ -1,26 +1,180 @@
...
-    </NavLink>
-
-    <NavLink exact to="/dashboard" className={styles.link} activeClassName={styles.active}>
-      Dashboard
-    </NavLink>
-  </Navbar>
-);
+type IHeaderProps = IunAuthProps | IAuthProps;
+
+interface IunAuthProps {
```
Comment (krzkaczor) [inline app/components/Header.tsx]: IUnAuthProps

File: app/components/Header.tsx [hunk 5]
```diff
@@ -1,26 +1,180 @@
...
+  name: string;
+  balanceEuro: number;
+  balanceNeu: number;
+}
+
+interface IHeaderState {
+  isOpen: boolean;
+}
+
+interface INeufundBrand {
```
Comment (krzkaczor) [inline app/components/Header.tsx]: please delete this interface and use className which is part of props for all components

File: app/components/Header.tsx [hunk 6]
```diff
@@ -1,26 +1,180 @@
...
+  balanceNeu: number;
+  name: string;
+  toggle: () => {};
+  isOpen: boolean;
+}
+
+export const NeufundBrand: React.SFC<INeufundBrand> = ({ classStyle }) => {
+  return (
+    <Row>
+      <img src="https://neufund.org/img/social_logo.png" className={styles.brand} alt="" />
```
Comment (krzkaczor) [inline app/components/Header.tsx]: please leave a todo note here that it needs to be taken from assets instead.

Comment (Moejoe90) [inline app/components/Header.tsx]: This will be changed when assets are added to repo

File: test/components/Header.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        <Authorized
+          name={name}
+          balanceEuro={balanceEuro}
+          balanceNeu={balanceNeu}
+          toggle={toggle}
+          isOpen={false}
+        />,
+      );
+      component.find(tid("button-toggle")).simulate("click");
+      expect(toggle.calledOnce).to.equal(true);
```
Comment (krzkaczor) [inline test/components/Header.spec.tsx]: use `expect(toggle)to.be.calledOnce`

Comment (krzkaczor) [reply test/components/Header.spec.tsx]: https://github.com/domenic/sinon-chai

File: app/components/Header.tsx (hunk 4 above)
Comment (Moejoe90) [reply app/components/Header.tsx]: done

File: app/components/Header.tsx (hunk 5 above)
Comment (Moejoe90) [reply app/components/Header.tsx]: done

Comment (Moejoe90) [reply app/components/Header.tsx]: className is not part of `React.SFC` so we will need an interface for it, however, i will change the name to `IReducedHTML` what do you think?

File: app/components/Header.tsx (hunk 6 above)
Comment (Moejoe90) [reply app/components/Header.tsx]: done image taken from asset and a `TODO` note is added

File: test/components/Header.spec.tsx (hunk 7 above)
Comment (Moejoe90) [reply test/components/Header.spec.tsx]: changed, It was first implemented as example available in  
http://airbnb.io/enzyme/docs/api/shallow.html

File: app/components/Header.module.scss [hunk 8]
```diff
@@ -1,16 +1,65 @@
...
+    border-bottom: 1.1rem;
+    border-bottom-style: solid;
+    border-bottom-color: $white;
+  }
+}
+
+.brand {
+  width: 3.75rem;
+  height: 1.875rem;
+  border-right: .09rem;
```
Comment (krzkaczor) [inline app/components/Header.module.scss]: all borders should have minimal width I think. According to https://offroadcode.com/rem-calculator/ it's: `1px = 0.0625rem`

File: app/components/Header.module.scss [hunk 9]
```diff
@@ -1,16 +1,65 @@
...
+}
+
+.active {
+  font-weight: bold;
+}
+
+@include media-breakpoint-up(sm) {
+  .active {
+    padding-bottom: .5rem;
+    border-bottom: 1.1rem;
```
Comment (krzkaczor) [inline app/components/Header.module.scss]: look: border note

File: app/components/Header.module.scss [hunk 10]
```diff
@@ -1,16 +1,65 @@
...
-.active {
-  font-weight: bold;
+.mini-header {
+  font-size: .75rem;
+  color: $white;
+}
+
+.title {
+  font-size: 1rem;
+  font-weight: 600;
```
Comment (krzkaczor) [inline app/components/Header.module.scss]: use font-weight predefined variables from bootstrap

File: app/components/Header.module.scss [hunk 11]
```diff
@@ -1,16 +1,65 @@
...
   &:hover {
     color: $white;
+    text-decoration: none;
   }
 }
 
-.active {
-  font-weight: bold;
+.mini-header {
+  font-size: .75rem;
```
Comment (krzkaczor) [inline app/components/Header.module.scss]: use predefined variables from bootstrap

File: test/components/Header.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    it("should render Authorized Navbar", () => {
+      const component = shallow(
+        <Header
+          isAuthorized={true}
+          name={name}
+          balanceEuro={balanceEuro}
+          balanceNeu={balanceNeu}
+        />,
+      );
+      expect(component.find(Authorized)).to.have.length(1);
```
Comment (krzkaczor) [inline test/components/Header.spec.tsx]: please assert here something like `<Authorized and all props here/>` so make sure that props are passed correctly

File: test/components/Header.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
+import { expect } from "chai";
+import { shallow } from "enzyme";
+import * as React from "react";
+import * as sinon from "sinon";
+import { Authorized, Header, UnAuthorized } from "../../app/components/Header";
+import { tid } from "../testUtils";
+
+describe("<Header />", () => {
+  const name = "moe";
```
Comment (krzkaczor) [inline test/components/Header.spec.tsx]: consider them putting into one object called defaultProps as we do it other tests

File: test/components/Header.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+  const balanceEuro = 0;
+  const balanceNeu = 0;
+  describe("<Header />", () => {
+    it("should render UnAuthorized Navbar", () => {
+      const component = shallow(<Header isAuthorized={false} />);
+      expect(component.find(UnAuthorized)).to.have.length(1);
```
Comment (krzkaczor) [inline test/components/Header.spec.tsx]: please assert here something like <UnAuthorized /> which makes sure that no props were passed to that component

**Comment by krzkaczor** [comment]
image
 it's broken on certain width

**Comment by krzkaczor** [comment]
image
what is the point of having a separate row with a spacer? couldn't it be solved by having margin-top and some border-top?

File: app/components/Header.module.scss (hunk 9 above)
Comment (Moejoe90) [reply app/components/Header.module.scss]: How would you style 
border

File: app/components/Header.module.scss (hunk 8 above)
Comment (Moejoe90) [reply app/components/Header.module.scss]: good note

File: app/components/Header.module.scss (hunk 9 above)
Comment (krzkaczor) [reply app/components/Header.module.scss]: i am sorry — this one's fine. I was referring to borders with minimal width. I think this is what you wanted to have with  `border-right: .09rem;` below

File: app/components/Header.module.scss (hunk 11 above)
Comment (Moejoe90) [reply app/components/Header.module.scss]: done, taken out of CSS and used `font-weight-bold` by bootstrap

File: app/components/Header.module.scss (hunk 9 above)
Comment (Moejoe90) [reply app/components/Header.module.scss]: Yeah for the one below definitely you on spot

File: app/components/Header.module.scss (hunk 10 above)
Comment (Moejoe90) [reply app/components/Header.module.scss]: Changed from `600` to `font-weight-bold` but take into note bootstrap has only four weights I think 
bold `700`
normal `500` 
if we want to implement weight 600 then we will have to manually add it
https://getbootstrap.com/docs/4.0/utilities/text/#font-weight-and-italics

**Comment by Moejoe90** [comment]
@krzkaczor 
The idea was for the separator to act as a different component that you can take off the view and grid if you wanted. I think this makes it simpler to change the spacer in the future to whatever we want. Based on designs without indulging too much into the complexities of defining many CSS styles and `@media` queries. 
If this is not the best approach. Then it can be amended to what you think is best.

**Comment by Moejoe90** [comment]
fixed layout
image

**Pull request merged**